src/service/poc_mobile.proto

```diff
@@ -38,4 +38,132 @@ service poc_mobile {
   rpc submit_speedtest(speedtest_req_v1) returns (speedtest_resp_v1);
   rpc submit_cell_heartbeat(cell_heartbeat_req_v1)
       returns (cell_heartbeat_resp_v1);
-}
\ No newline at end of file
+}
+
+message file_info {
+  string key = 1;
+  file_type file_type = 2;
+  uint64 timestamp = 3;
+  uint64 size = 4;
+}
+
+message processed_files { repeated file_info files = 1; }
+
+message share {
+  string cbsd_id = 1;
+  bytes pub_key = 2;
+  uint64 weight = 3;
+  uint64 timestamp = 4;
+  cell_type cell_type = 5;
+}
+
+message shares { repeated share shares = 1; }
+
+message invalid_share {
+  string cbsd_id = 1;
+  bytes pub_key = 2;
+  uint64 weight = 3;
+  uint64 timestamp = 4;
+  cell_type cell_type = 5;
+  string invalid_reason = 6;
+}
+
+message invalid_shares { repeated invalid_share shares = 1; }
+
+message speed_share {
+  bytes pub_key = 1;
+  /// bytes/sec
+  uint64 upload_speed = 2;
+  /// bytes/sec
+  uint64 download_speed = 3;
+  uint32 latency = 4;
+  uint64 timestamp = 5;
+}
+
+message speed_share_list {
+  bytes gw_pub_key = 1;
+  repeated speed_share speed_shares = 2;
+}
+
+message speed_shares { repeated speed_share_list speed_shares = 1; }
+
+message invalid_speed_share {
```

Comment:
do we need both valid and invalid types for speed shares when the only difference is the addition of the `reason` string on the invalid side?
any reason not to combine them, possibly with a `is_valid` boolean field?

Reply:
I think both ways are valid and it just comes down to personal preference. However, I will say that we are likely to only put the valid protos in buckets that will be inputs to other services, so I'd prefer those services to not have to potentially deal with "is_invalid" as a required false value

Reply:
that depends on what you consider "other services"; we'll need to expose reports of both valid _and_ invalid reward inputs for accountability so _some_ services will be consuming the invalid speed shares and heartbeats even if it's not the rewards servers

Reply:
I'll go ahead and combine as many as I can - I do need to give invalid_speed_shares as a separate list because they may not have a valid gw_pub_key